src/app/auth/signup.tsx

```diff
@@ -1,25 +1,92 @@
-import { Redirect, Link, router } from 'expo-router';
+import { Link, router } from 'expo-router';
 import React, { useState } from 'react';
-import { Alert, View } from 'react-native';
-import { Button, Input } from 'react-native-elements';
+import { Alert, Text, View, StyleSheet } from 'react-native';
+import { SafeAreaView } from 'react-native-safe-area-context';
+import validator from 'validator';
 
+import StyledButton from '../../components/StyledButton/StyledButton';
+import UserStringInput from '../../components/UserStringInput/UserStringInput';
 import globalStyles from '../../styles/globalStyles';
 import { useSession } from '../../utils/AuthContext';
+import supabase from '../../utils/supabase';
+import { Icon } from 'react-native-elements';
 
 function SignUpScreen() {
-  const { session, signUp } = useSession();
+  const { signUp } = useSession();
 
+  const [usernameError, setUsernameError] = useState('');
+  const [emailError, setEmailError] = useState('');
+  const [username, setUsername] = useState('');
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
   const [email, setEmail] = useState('');
   const [password, setPassword] = useState('');
+  const [passwordTextHidden, setPasswordTextHidden] = useState(true);
   const [loading, setLoading] = useState(false);
+  const validUsernameCharacters = /^\w+$/g;
 
-  if (session) {
-    return <Redirect href="/home" />;
-  }
+  const setAndCheckUsername = async (newUsername: string) => {
```

Comment:
Edit this function such that when the input is empty, there are no error messages shown. Right now, if I type in an invalid username and then delete it, the error message from the invalid username is still shown when the field is empty.

Reply:
Also seeing this same issue with the email text input, so make sure to update that function as well